src/components/explorer/IndividualTracks.js

```diff
@@ -197,67 +291,76 @@ const IndividualTracks = ({ individual }) => {
+                "igvBrowserRef.current =", igvBrowserRef.current);
+            return cleanup;
         }
 
-        const indexedTracks = allFoundFiles.filter(
-            (t) => t.viewInIgv && igvUrls[t.filename].dataUrl && igvUrls[t.filename].indexUrl,
-        );
-
-        const unindexedTracks = allFoundFiles.filter((t) => t.viewInIgv && igvUrls[t.filename].url);
-
-        const igvIndexedTracks = indexedTracks.map((t) => ({
-            format: t.file_format,
-            url: igvUrls[t.filename].dataUrl,
-            indexURL: igvUrls[t.filename].indexUrl,
-            name: t.filename,
-            squishedCallHeight: SQUISHED_CALL_HEIGHT,
-            expandedCallHeight: EXPANDED_CALL_HEIGHT,
-            displayMode: DISPLAY_MODE,
-            visibilityWindow: VISIBILITY_WINDOW,
-        }));
-
-        const igvUnindexedTracks = unindexedTracks.map((t) => ({
-            format: t.file_format,
-            url: igvUrls[t.filename].url,
-            name: t.filename,
-            squishedCallHeight: SQUISHED_CALL_HEIGHT,
-            expandedCallHeight: EXPANDED_CALL_HEIGHT,
-            displayMode: DISPLAY_MODE,
-            visibilityWindow: VISIBILITY_WINDOW,
-        }));
-
-        const igvTracks = igvUnindexedTracks.concat(igvIndexedTracks);
+        setCreatingIgvBrowser(true);
+
+        const initialIgvTracks = allFoundFiles
+            .filter((t) => t.viewInIgv && t.genome_assembly_id === selectedAssemblyID && igvUrls[t.filename].url)
+            .map((t) => buildIgvTrack(igvUrls, t));
 
         const igvOptions = {
-            genome: genome,
+            genome: availableBrowserGenomes[selectedAssemblyID],
             locus: igvPosition,
-            tracks: igvTracks,
+            tracks: initialIgvTracks,
         };
 
-        igv.createBrowser(igvRef.current, igvOptions).then((browser) => {
-            igv.browser = browser;
-            igvRendered.current = true;
+        console.debug("creating igv.js browser with options:", igvOptions, "; tracks:", initialIgvTracks);
 
-            igv.browser.on(
+        igv.createBrowser(igvDivRef.current, igvOptions).then((browser) => {
+            browser.on(
                 "locuschange",
                 debounce((referenceFrame) => {
                     storeIgvPosition(referenceFrame);
                 }, DEBOUNCE_WAIT),
             );
+            igvBrowserRef.current = browser;
+            setCreatingIgvBrowser(false);
+            console.debug("created igv.js browser instance:", browser);
+        }).catch((err) => {
+            message.error(`Error creating igv.js browser: ${err}`);
+            console.error(err);
         });
-    }, [igvUrls]);
+
+        return cleanup;
+
+        // Use viewableResults as the track dependency, not allFoundFiles, since allFoundFiles is regenerated if a
+        // track's visibility changes
+    }, [igvUrls, viewableResults, availableBrowserGenomes, selectedAssemblyID]);
```

Comment:
allFoundFiles is left out on purpose (see the comment, but I'll make it a bit clearer); refs cannot be used as dependencies since they never change memory address by design (it's a constant object whos internal reference, i.e., `.current`, changes what it points to)